### PR TITLE
Fixes https://github.com/csarofeen/pytorch/issues/292

### LINKED ISSUE #292: Lowering twice triggers assert
```.cpp
Fusion fusion;
    FusionGuard fg(&fusion);

    int y = 6, z = 8;
    DataType dtype = DataType::Float;
    auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);

    auto tv0 = makeDummyTensor(1);
    auto tv1 = makeDummyTensor(2);
    fusion.addInput(tv0);
    fusion.addInput(tv1);

    auto tv2 = broadcast(tv0, {true, false});
    auto tv3 = add(tv2, new Float(1.0));
    auto tv4 = add(tv3, tv1);

    fusion.addOutput(tv4);

    tv4->merge(0);

    tv4->split(0, 128);
    tv4->split(0, 4);

    tv2->computeAt(tv4, 1);

    fusion.printKernel();

    torch::jit::fuser::cuda::FusionExecutor fe;

    at::Tensor t0 = at::randn({z}, options).fill_(1.0);
    at::Tensor t1 = at::randn({y, z}, options).fill_(3.0);

    fe.compileFusion(&fusion);
    auto outputs = fe.runFusion({t0, t1});

    auto t3 = t0.add(1.0);
    auto t4 = t3.add(t1);

    TORCH_CHECK(t4.allclose(outputs[0]));
```
produces:
```
C++ exception with description "fuser_tv_ != nullptr INTERNAL ASSERT FAILED at "../torch/csrc/jit/codegen/cuda/kernel_ir.h":318, please report a bug to PyTorch. 
Exception raised from fuserTv at ../torch/csrc/jit/codegen/cuda/kernel_ir.h:318 (most recent call first):
```

### PR DESCRIPTION
**Pull request by csarofeen**
Fixes: https://github.com/csarofeen/pytorch/issues/292
CC: @tlemo

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/kernel_ir.cpp [hunk 1]
```diff
@@ -173,18 +173,16 @@ std::vector<IterDomain*> TensorDomain::noBroadcasts(
   return no_broadcast_domains;
 }
 
-TensorView::TensorView(TensorDomain* domain, DataType dtype)
-    : Val(ValType::KirTensorView, dtype, true, true), domain_(domain) {}
-
 TensorView::TensorView(const fuser::TensorView* tv) : Val(tv), fuser_tv_(tv) {
-  domain_ = lowerValue(tv->domain())->as<TensorDomain>();
+  domain_ = lowerValue(tv->domain())->as<kir::TensorDomain>();
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/kernel_ir.cpp]: please remove `kir::` (not needed, we're already in the kir namespace)

File: torch/csrc/jit/codegen/cuda/kernel_ir.cpp [hunk 2]
```diff
@@ -173,18 +173,16 @@ std::vector<IterDomain*> TensorDomain::noBroadcasts(
...
   memory_type_ = tv->getMemoryType();
 }
 
 TensorView::TensorView(const TensorView* src, IrCloner* ir_cloner)
     : Val(src, ir_cloner),
       domain_(ir_cloner->clone(src->domain_)),
-      memory_type_(src->memory_type_) {}
+      memory_type_(src->memory_type_),
+      fuser_tv_(src->fuser_tv_) {}
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/kernel_ir.cpp:185]: good catch!

File: torch/csrc/jit/codegen/cuda/kernel_ir.h [hunk 3]
```diff
@@ -302,8 +302,6 @@ class TORCH_CUDA_API TensorDomain : public Val {
 
 class TORCH_CUDA_API TensorView : public Val {
  public:
-  TensorView(TensorDomain* domain, DataType dtype);
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/kernel_ir.h:305]: nice

**Review by tlemo — APPROVED** [review]

File: torch/csrc/jit/codegen/cuda/kernel_ir.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/kernel_ir.cpp]: There's a few `kir::` instances in this file, might want to look through them.

**Pull request merged**